waiter/src/waiter/process_request.clj

```diff
@@ -229,11 +230,12 @@
        :fold-chunked-response? true
        :fold-chunked-response-buffer-size output-buffer-size
        :follow-redirects? false
-       :idle-timeout idle-timeout})))
+       :idle-timeout idle-timeout
+       :query-string query-params})))
```

Comment:
Will this ultimately end up encoding `"?q=a|b"` into `"?q=a%7Cb"` when sending to the backend?  We should avoid changing the request.

A different approach might be add a new key `raw-query-string` to jet's `request` parameter map and use it to override `Request.getQuery()` on the object returned by `HttpClient.newRequest`.  This could avoid the encoding happening in `Request.param`, which is ultimately what jet currently passes `query-string` to.

Reply:
Do we have to implement all of those?  I have a suspicion that only getQuery is used when rendering the bytes sent across the wire.

Reply:
If it does use `getURL` or `getParams` to write the request, then agreed that this approach won't work.

Reply:
All the `newRequest` methods eventually rely on a `URI` being passed. The implementation, `org.eclipse.jetty.client.HttpRequest`, constructor also expects a `URI` and is not easily extendible.

Creating a URI ends up encoding the query string:

```
(URI. "http", nil, "/path" "a=b|c", nil)
=> #object[java.net.URI 0x3180f50d "http:/path?a=b%7Cc"]
```

I'm not sure it is worth the effort to avoid encoding the query string.